frontend/src/main/pages/Shift/ShiftIndexPage.js

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { useBackend } from 'main/utils/useBackend';
+import BasicLayout from 'main/layouts/BasicLayout/BasicLayout';
+import ShiftTable from 'main/components/Shift/ShiftTable';
+import { Button } from 'react-bootstrap';
+import { useCurrentUser, hasRole } from 'main/utils/currentUser';
+
+export default function ShiftIndexPage() {
+
+    const currentUser = useCurrentUser();
+
+    const { data: shift, error: _error, status: _status } =
+    useBackend(
+        // Stryker disable all : don't test internal caching of React Query
+        ["/api/shift/all"],
+        { method: "GET", url: "/api/shift/all" },
+        []
+        // Stryker restore all
+    );
+
+    const createButton = () => {
+        if (hasRole(currentUser, "ROLE_ADMIN")) {
+            return (
+                <Button
+                    variant="primary"
+                    href="/shift/create"
+                    style={{ float: "right" }}
+                >
+                    Create Shift
+                </Button>
+            );
+        }
+    };
+    
+    return (
+        <BasicLayout>
+            <div className="pt-2">
+                {createButton()}
```

Comment:
Never call components directly. This will violate rules of hooks if the component uses hooks and you conditionally render it. I suggest inlining the render here.